keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/HalfCauchyVertex.java

```diff
@@ -49,6 +52,29 @@ public double logProb(DoubleTensor value) {
         return Double.NEGATIVE_INFINITY;
     }
 
+    @Override
+    public LogProbGraph logProbGraph() {
+        final DoublePlaceholderVertex xPlaceholder = new DoublePlaceholderVertex(this.getShape());
+        final DoublePlaceholderVertex locationPlaceholder = new DoublePlaceholderVertex(getLocation().getShape());
+        final DoublePlaceholderVertex scalePlaceholder = new DoublePlaceholderVertex(getScale().getShape());
+
+        final DoubleVertex cauchyLogProbOutput = Cauchy.logProbOutput(xPlaceholder, locationPlaceholder, scalePlaceholder);
+
+        final DoubleVertex result = cauchyLogProbOutput.plus(LOG_TWO);
+        final DoubleVertex invalidMask = xPlaceholder.toLessThanMask(LOC_ZERO);
+        final DoubleVertex halfCauchyLogProbOutput = result.setWithMask(invalidMask, Double.NEGATIVE_INFINITY);
+
+        // Set the value of locationPlaceholder since we know it's 0.
+        locationPlaceholder.setValue(LOC_ZERO);
```

Comment:
Can the place holder just be a constant?

Reply:
Hmmm.. would that connect all computation graphs to that vertex?